### DISCUSSION TITLE
can I get a `Res` from a `World`?

### DISCUSSION THREAD

**Discussion by atornity** [bevyengine/bevy#11552] [Q&A] [answered]
I'm making a thingy where I can add `Modifiers` to components and I want to be able to write stuff like

```rust
// where Health is a component and Score is a resource
let health_modifier = Modifier::new(|val: &mut Health, speed: &Speed, score: Res<Score>| {
    val.0 = speed.0 + score.0;
});
// this runs whenever Speed, Score or Base<Health> changes.
```

ok, the example doesn't make any sense but I want the first argument to be a specific component then every argument after that can be anything that implements `ModParam` which has a method for getting the stuff from the `World`.

So basically, I need a way of getting a `Res` from a `World` for this to work.

**Comment by alice-i-cecile** [comment, marked as answer]
You want one of the methods on `World` like [`World::resource`](https://docs.rs/bevy/latest/bevy/ecs/world/struct.World.html#method.resource). 

If you run into troubles with the borrow checker, see [`resource_scope`](https://docs.rs/bevy/latest/bevy/ecs/world/struct.World.html#method.resource_scope), [`SystemState`](https://docs.rs/bevy/latest/bevy/ecs/system/struct.SystemState.html) or even [one shot systems](https://github.com/bevyengine/bevy/blob/main/examples/ecs/one_shot_systems.rs).

**Reply by atornity** [reply]
i can't use `World::resource` since it returns `&T`, not `Res<T>` but I'll check out `SystemState` tomorrow. Thanks!
